### ISSUE TITLE
Unable to resolve domain names after waking up from suspension

### ISSUE THREAD

**Issue by k4yt3x** [DNSCrypt/dnscrypt-proxy#432]
When I wake my laptop up from suspension and connect to a new WiFi using NetworkManager, I have to do
```
$ sudo systemctl restart dnscrypt-proxy
```
to make the resolving service usable again.

If I do
```
$ nslookup google.com 1.1.1.1
```
It works and gives me the IP addrs.

However, if I do
```
$ nslookup google.com 127.0.2.1
```
without restarting the service, it won't work.

Version `2.0.11` running on `4.15.0-kali3-amd64 #1 SMP Debian 4.15.17-1kali1 (2018-04-25) x86_64 GNU/Linux`

**Comment by jedisct1** [comment]
If you are using systemd sockets, can you try without them, using standard sockets managed by the proxy?

**Comment by evilvibes** [comment]
I'm also seeing this on macOS when using my servers when i use say cloudflare there isn't a issue. I think it has something to do with using the dnscrypt-server-docker and the dnscrypt-proxy together. I have to restart dnscrypt-proxy to get it to work again.

Maybe you could test this out? @jedisct1

**Comment by martijnkruiten** [comment]
I too have this issue and this issue also extends to the cloudflared dns daemon. I just switched to the socket to see if that helps (even though @jedisct1 suggested the opposite). I won't be testing it right away, but I'll report back once I've resumed my laptop from suspension.

**Comment by k4yt3x** [comment]
The issue remiains. Dnscrypt doesn't seems to realize that a new network is connected and remains hibernating.

Aside from that, when I restart the service using systemd after connecting to a wifi the resolver starts working again. I watched the syslog and here are two lines that caught my attention. I tried downloading the two files and put them in the directory manually. It worked fine the first time but then dnscrypt insists downloading a newer file.
~~~
Jun  5 13:27:00 KARDINAL dnscrypt-proxy[10953]: [2018-06-05 13:27:00] [WARNING] /etc/dnscrypt-proxy/public-resolvers.md: open sf-i7fk3apeayl6u6z3.tmp: read-only file system
Jun  5 13:27:00 KARDINAL dnscrypt-proxy[10953]: [2018-06-05 13:27:00] [WARNING] /etc/dnscrypt-proxy/public-resolvers.md.minisig: open sf-ywxa7jw7rta7pmsd.tmp: read-only file 
~~~

Maybe this will help.
